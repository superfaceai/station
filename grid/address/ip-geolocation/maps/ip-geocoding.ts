/* eslint-disable jest/no-export */
import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

export const ipGeolocationTest = (
  provider: string,
  recordingOptions?: RecordingProcessOptions
): void => {
  describe(`address/ip-geolocation/${provider}`, () => {
    let superfaceTest: SuperfaceTest;

    beforeEach(() => {
      superfaceTest = buildSuperfaceTest({
        profile: 'address/ip-geolocation',
        provider,
      });
    });

    describe('IpGeolocation', () => {
      describe('when api key is valid', () => {
        it('should return geolocation coordinates and address for valid IP address', async () => {
          await expect(
            superfaceTest.run(
              {
                useCase: 'IpGeolocation',
                input: { ipAddress: '8.8.8.8' },
              },
              recordingOptions
            )
          ).resolves.toMatchSnapshot();
        });

        it('should return client IP geolocation coordinates when no IP address specified in input', async () => {
          await expect(
            superfaceTest.run(
              {
                useCase: 'IpGeolocation',
                input: {},
              },
              recordingOptions
            )
          ).resolves.toMatchSnapshot();
        });

        it('should return bad request error when IP address format is wrong', async () => {
          await expect(
            superfaceTest.run(
              {
                useCase: 'IpGeolocation',
                input: { ipAddress: '256.256.256.256' },
              },
              recordingOptions
            )
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
