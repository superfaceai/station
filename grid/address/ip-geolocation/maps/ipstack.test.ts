import { RecordingDefinitions } from '@superfaceai/testing';

import { ipGeolocationTest } from './ip-geocoding';

const IPSTACK_API_KEY_REDACTED_VALUE = 'IPSTACK_API_KEY_REDACTED';

ipGeolocationTest('ipstack', {
  //Testing library is not processing recordings because integration parameter is used to replace HTTP scheme which is not supported by testing lib
  processRecordings: false,
  beforeRecordingSave: (recordings: RecordingDefinitions) => {
    for (const recording of recordings) {
      if (process.env['IPSTACK_API_KEY']) {
        recording.path = recording.path.replace(
          process.env['IPSTACK_API_KEY'],
          IPSTACK_API_KEY_REDACTED_VALUE
        );
      }
    }
  },
  beforeRecordingLoad: (recordings: RecordingDefinitions) => {
    for (const recording of recordings) {
      if (process.env['IPSTACK_API_KEY']) {
        recording.path = recording.path.replace(
          IPSTACK_API_KEY_REDACTED_VALUE,
          process.env['IPSTACK_API_KEY']
        );
      }
    }
  },
});
