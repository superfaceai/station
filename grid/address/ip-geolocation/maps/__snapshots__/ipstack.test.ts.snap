// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`address/ip-geolocation/ipstack IpGeolocation when api key is valid should return bad request error when IP address format is wrong 1`] = `
Err {
  "error": "MappedHTTPError: Expected HTTP error
Properties: {
  \\"title\\": \\"Bad request\\",
  \\"detail\\": \\"The IP Address supplied is invalid.\\"
}
AST Path: definitions[0].statements[2].responseHandlers[0].statements[3]
Original Map Location: Line 24, column 7",
}
`;

exports[`address/ip-geolocation/ipstack IpGeolocation when api key is valid should return client IP geolocation coordinates when no IP address specified in input 1`] = `
Ok {
  "value": Object {
    "addressCountry": "Czechia",
    "addressCountryCode": "CZ",
    "addressLocality": "Jablonec nad Nisou",
    "addressRegion": "Libereck√Ω kraj",
    "ipAddress": "94.127.131.198",
    "latitude": 50.68013000488281,
    "longitude": 15.321399688720703,
    "postalCode": "468 22",
    "timeZone": undefined,
  },
}
`;

exports[`address/ip-geolocation/ipstack IpGeolocation when api key is valid should return geolocation coordinates and address for valid IP address 1`] = `
Ok {
  "value": Object {
    "addressCountry": "United States",
    "addressCountryCode": "US",
    "addressLocality": "Mountain View",
    "addressRegion": "California",
    "ipAddress": "8.8.8.8",
    "latitude": 37.38801956176758,
    "longitude": -122.07431030273438,
    "postalCode": "94041",
    "timeZone": undefined,
  },
}
`;
